### PR TITLE
untar_data rename issue

### PR DESCRIPTION
**Pull request by tcapelle**
Update untar_data to rename destination folder if it is not equal to download archive

### CODE REVIEW

**Comment by tcapelle** [comment]
I don't get any diff...

**Comment by sgugger** [comment]
Are you sure you have the latest nbdev? From all the diff I see, I think you need to update it then rebuild the library.

File: fastai2/data/external.py [hunk 1]
```diff
@@ -243,4 +249,6 @@ def untar_data(url, fname=None, dest=None, c_key='data', force_download=False, e
         if _get_check(url) and _check_file(fname) != _get_check(url):
             print(f"File downloaded is broken. Remove {fname} and try again.")
         extract_func(fname, dest.parent)
+        print(f'fname: {fname}, dest: {dest}')
```
Comment (sgugger) [inline fastai2/data/external.py:252]: We try to avoid being verbose. Maybe just make a test to see if a rename is necessary and just print something then?

**Comment by sgugger** [comment]
Thanks for fixing the build. I just have one comment on the design, if you can address it.

File: fastai2/data/external.py (hunk 1 above)
Comment (tcapelle) [reply fastai2/data/external.py:252]: Oh I forgot to remove this...

Comment (tcapelle) [reply fastai2/data/external.py]: I refactored the code, to check outside, I added a test. It is complicated to add an integral test without a missnamed downloadable dataset.
It is so painful that the unzip and untar methods don't give back the extrated folder.

File: nbs/04_data.external.ipynb [hunk 2]
```diff
@@ -228,10 +228,10 @@
     {
```
[the remark this answers is not in the document]
Comment (tcapelle) [reply nbs/04_data.external.ipynb:228]: ohh

[the remark this answers is not in the document]
Comment (tcapelle) [reply nbs/04_data.external.ipynb:228]: done

**Comment by sgugger** [comment]
Thanks a lot!

**Pull request merged**